Neos.ContentRepository.Core/Classes/Projection/ContentGraph/NodePath.php

```diff
@@ -77,32 +81,37 @@ public function appendPathSegment(NodeName $nodeName): self
     }
 
     /**
-     * @return NodeName[]
+     * @return array<int,NodeName>
      */
     public function getParts(): array
     {
+        if ($this->isRoot()) {
+            return [];
+        }
         $pathParts = explode('/', ltrim($this->value, '/'));
         return array_map(static fn (string $pathPart) => NodeName::fromString($pathPart), $pathParts);
     }
 
     public function getDepth(): int
     {
-        if (!$this->isAbsolute()) {
-            throw new \RuntimeException(sprintf(
-                'Depth of relative node path "%s" cannot be determined',
-                $this->value
-            ), 1548162166);
-        }
-        return count($this->getParts());
+        throw new \RuntimeException(sprintf(
+            'Depth of relative node path "%s" cannot be determined',
+            $this->value
+        ), 1548162166);
     }
```

Comment:
makes sense, but we might aswell remove it?

Reply:
yep, simply forgot about it; It's dead code und must be gone. It's kind of breaking though, see below

Reply:
done